### DISCUSSION TITLE
Using same input video, can we use previously generated occlusion masks?

### DISCUSSION THREAD

**Discussion by bigmanong** [volotat/SD-CN-Animation#138] [Ideas]
I have been playing with this over the past few days, and I really like the outputs I am getting.  However, I am wondering if there would be a way to add a setting that allows you to use the preprocessed images from a previous vid2vid generation to create a new video with a different prompt?  This would allow you to skip about half of the work if you are doing several prompts for the same input video.

Combined with [Oxencorp's post](https://github.com/volotat/SD-CN-Animation/discussions/136) about batch prompts, this would be amazing.

**Comment by aulerius** [comment]
Yes! And ideally import of not just previous, but externally generated data (from 3d software)!
Related to my proposal here: #130

**Comment by peteTater** [comment]
Been thinking about this a lot. I'm testing an idea today, downloading each occlusion frame of the sequence as it generates. Hoping if I take the occlusion frame sequence, create a copy of it on the timeline of a vid editor, reverse the copy to create a loop, then feed the video output back in to txt2vid as a guide to see if I can make loops that look more natural than just doing a reverse vid of the identical frames from the original final output sequence. Even if it works, seamless looping hinges on whether the "Apply color correction to img2img results to match original colors" from the Settings tab actually regulates the colours in order for the first and last frames of the sequence to have a similar palette. So far with the prompts I've been using, this has never quite worked as hoped. Will report back on whether I have any success with it.

**Reply by peteTater** [reply]
Spoke too soon, had assumed I could just use the little download button in the top right of the occlusion frame window to download every frame as they're generated, but it only allows to d/L a single image after generation completes. Hopefully the frames are temporarily cached to disk during generation.

Edit: Popping the hood on txt2vid.py I think getting the "occ" frames written to disk might be something my Python literate waifu might be able to help me take a stab at this eve.
